components/web/services/org.wso2.carbon.apimgt.authenticator/src/main/java/org/wso2/carbon/apimgt/authenticator/LoginTokenService.java

```diff
@@ -108,4 +121,48 @@ public void revokeAccessToken(String appName, String accessToken) throws KeyMana
             return AuthUtil.getConsumerKeySecretMap().get(appName);
         }
     }
+
+    private String getUsernameFromJWT(String jwt) throws KeyManagementException {
+        if (jwt != null && jwt.contains(".")) {
+            String[] jwtParts = jwt.split("\\.");
+            JWTTokenPayload jwtHeader = new Gson().fromJson(new String(Base64.getDecoder().decode(jwtParts[1]),
+                    StandardCharsets.UTF_8), JWTTokenPayload.class);
```

Comment:
Is charset will be always utf8? or should we take the underline system charser of jvm?